examples/multi-threads-1.cpp

```diff
@@ -0,0 +1,163 @@
+//
+//
+// Copyright (c) 2017 Ivan Baidakou (basiliscos) (the dot dmol at gmail dot com)
+//
+// Distributed under the MIT Software License
+
+// this is an example how to use bredis in multi-threaded environment, i.e
+// the access to sockets and buffers should be routed via asio::io_context::strand
+//
+// sample output:
+//
+// connecting to 127.0.0.1:6379
+// connected
+// pings: 37582, pongs: 37264
+// pings: 70115, pongs: 69801
+// pings: 109627, pongs: 109627
+// pings: 151142, pongs: 151142
+// pings: 194718, pongs: 194521
+// pings: 233520, pongs: 233519
+// pings: 272180, pongs: 272179
+// pings: 310104, pongs: 310103
+// pings: 346666, pongs: 346666
+// pings: 385500, pongs: 385500
+
+
+#include <boost/thread.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/program_options.hpp>
+#include <boost/tokenizer.hpp>
+
+#include <bredis.hpp>
+//#include "../t/SocketWithLogging.hpp"
+
+// alias namespaces
+namespace r = bredis;
+namespace asio = boost::asio;
+namespace po = boost::program_options;
+namespace sys = boost::system;
+
+using socket_t = asio::ip::tcp::socket;
+using next_layer_t = socket_t;
+//using next_layer_t = r::test::SocketWithLogging<asio::ip::tcp::socket>;
```

Comment:
This is forked from a public repo and not part of the changes Vlad made so I think we should avoid changing impl./example files from the host repo. Since if we upgrade this library from the source we would also have to look at merge conflicts. Same comment applies to below suggestions